hibernate-core/src/main/java/org/hibernate/sql/results/graph/embeddable/AbstractEmbeddableInitializer.java

```diff
@@ -269,7 +268,7 @@ private void prepareCompositeInstance(RowProcessingState processingState) {
 		);
 	}
 
-	private boolean isParentInstanceNull() {
+	private boolean shouldSkipInitilization(){
```

Comment:
It's a bit odd that the method now changes the state. Maybe name it `resolveShouldSkipInitialization` instead to signal that something is happening?